app/src/components/main/attributeList/editableAttribute/attributeEditor/attributeEditor.jsx

```diff
@@ -94,52 +84,86 @@ export class AttributeEditor extends Component {
   formatValue = (value) => (value ? { value, label: value } : null);
   parseValue = (value) => (value ? value.value : undefined);
 
+  handleKeyChange = (key) => {
+    this.setState({
+      key: this.parseValue(key),
+      errors: this.getValidationErrors(this.parseValue(key), this.state.value),
```

Comment:
You use state variables for calculating the new state. Due to the fact that React may batch setState calls, it would be better to use a function instead of an object in setState call.